### DISCUSSION TITLE
vim.opt.showtabline = 0 inside options.lua file is not disabling tabline

### DISCUSSION THREAD

**Discussion by morenohernan** [LazyVim/LazyVim#1429] [Q&A] [answered]
Hi, 
I like some of the bufferline functionalities but I don't like to the tabline (with the tabs), so I've tried this sentence in options.lua
vim.opt.showtabline = 0

However it seems is not taking any effect, in the default behavior ... Any hint?

**Comment by dpetka2001** [comment, marked as answer]
It probably is some plugin that overrides it. You could create an autocommand to enforce it, like so
```lua
vim.api.nvim_create_autocmd("BufReadPre", {
  group = vim.api.nvim_create_augroup("showtabline", { clear = true }),
  callback = function()
    vim.opt.showtabline = 0
  end,
  desc = "Set tabline to 0",
})
```

**Reply by morenohernan** [reply]
Thanks @dpetka2001 , it worked.

**Comment by emseidov** [comment]
The accepted answer no longer seems to work, BTW. At least it didn't work for me. This is what worked instead:

```lua
vim.api.nvim_create_autocmd("User", {
  callback = function()
    vim.opt.showtabline = 0
  end,
})
```

I tried to make it work without autocmd, by setting bufferline's `auto_toggle_bufferline` option to `false`, because it seems to enable `showtabline` flag even when it's set 0, but it didn't work for some reason. I'll try to debug it when I have more time.

**Reply by ZachVec** [reply]
why not simply `vim.o.showtabline = 0` in options.lua?

**Reply by emseidov** [reply]
It get's overwritten by some other plugin.

**Reply by dpetka2001** [reply]
It's the plugin `bufferline.nvim` that sets the option. Open Neovim with `nvim -V1 init.lua` and then type `:verbose set showtabline?` to see where it was last set. In a fresh LazyVim installation it will show you that it gets set by `bufferline.nvim`.

You can do 
```lua
return {
  {
    "akinsho/bufferline.nvim",
    -- enabled = false,
    opts = {
      options = {
        auto_toggle_bufferline = false,
      },
    },
  },
}
```
to disable the ability of `bufferline.nvim` overwriting `showtabline` setting or you can also uncomment `enabled = false` to entirely disable the plugin if you don't use it.

**Reply by emseidov** [reply]
I tried the `auto_toggle_bufferline` option, but it didn't work for me for some reason... Tried to debug it for a bit, but gave up later.

**Reply by dpetka2001** [reply]
I tried that setting with a default LazyVim installation and it works as expected. Are you sure you tried the exact same block of code (just copy/paste it to be sure)? 

And like I said in my previous comment, you can open Neovim with `nvim -V1 init.lua` and then inside do `:verbose set showtabline?` and it will show you where its value was last set.

**Reply by emseidov** [reply]
@dpetka2001 Okay, had some free time and checked it again now. It works as expected with the default config, but there's an issue when using mini-starter (snacks dashboard replacement in extras), which I'm using.

The toggle state of tabline in which-key doesn't seem to get updated, or set correctly on startup. Which is why I was forced to use autocmds instead `auto_toggle_bufferline`, because I want to be able to toggle tabline when needed, but I can't when I'm using mini-starter.
 
 Any idea how can I solve this, BTW? I feel like I have to update the tabline toggle state in which-key on startup.

**Reply by dpetka2001** [reply]
That's because `mini.starter` explicitly sets `showtabline` to 1 [here](https://github.com/echasnovski/mini.starter/blob/aa1f05d40e59ddc652a7333a05e707bd9233b9d0/lua/mini/starter.lua#L1371) and LazyVim defines the mapping [here](https://github.com/LazyVim/LazyVim/blob/25abbf546d564dc484cf903804661ba12de45507/lua/lazyvim/config/keymaps.lua#L148). 

You see that LazyVim first uses the value of `showtabline` if it's greater than 0 and if it's not it defaults to 2. So, in the case of `mini.starter` it will always be 1 so it will never be set to 2. And the default Neovim behavior when the value is set to 1, is to only show the tabline if there are at least 2 tab pages.

You can set in your configuration in `~/.config/nvim/lua/config/keymaps.lua` the following
```lua
Snacks.toggle.option("showtabline", { off = 0, on = 2, name = "Tabline" }):map("<leader>uA")
```
and it should work.

**Reply by emseidov** [reply]
It works, and honestly, you're amazing! 🙂 Thanks for taking the time to give a detailed answer (including plugin links is so neat).
